component/loki/source/api/internal/lokipush/push_api_server_test.go

```diff
@@ -0,0 +1,267 @@
+package lokipush
+
+// This code is copied from Promtail (3478e180211c17bfe2f3f3305f668d5520f40481) with changes kept to the minimum.
+// The lokipush package is used to configure and run the HTTP server that can receive loki push API requests and
+// forward them to other loki components.
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component/common/loki/client/fake"
+	fnet "github.com/grafana/agent/component/common/net"
+	frelabel "github.com/grafana/agent/component/common/relabel"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/grafana/dskit/flagext"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/client"
```

Comment:
Let's avoid importing promtail directly and use the types in component/common/loki instead, like we did for the component code.

Reply:
Yes! sorry, I missed this when I re-copied this test.